common/djangoapps/third_party_auth/tests/test_pipeline_integration.py

```diff
@@ -338,6 +346,40 @@ def test_provider_settings_redirect_to_registration(self, send_to_registration_f
                 assert response.status_code == 302
                 assert response.url == expected_redirect_url
 
+    @ddt.data(
+        ('non_existing_user_email@example.com', '/register'),
+        ('email@example.com', '/login'),
+        (None, '/register')
+    )
+    @ddt.unpack
+    def test_redirect_for_saml_based_on_email_only(self, email, expected_redirect_url):
+        """
+        Test that if the provider is saml, then only email(and not username)
+        is used to determine if a user already exists.
+        """
+        saml_provider = mock.MagicMock(
+            slug='unique_slug',
+            skip_email_verification=False
+        )
+        with mock.patch('third_party_auth.pipeline.provider.Registry.get_from_pipeline') as get_from_pipeline:
+            get_from_pipeline.return_value = saml_provider
+            with mock.patch(
+                'third_party_auth.pipeline.provider.Registry.get_enabled_by_backend_name'
+            ) as enabled_saml_providers:
+                enabled_saml_providers.return_value = [saml_provider, ]
```

Comment:
Should we also check other providers to be on a safe side?

Reply:
Can you elaborate? ... thanks